src/device.cpp

```diff
@@ -195,6 +195,7 @@ device::device(std::shared_ptr<context> ctx,
 
 device::~device()
 {
+    _context->stop_device_watcher();
```

Comment:
Seems like it'd work, sure -- but why stop the context picking up new devices?

Reply:
@v-lopez Can you suggest a better solution to this issue?

Reply:
Haven't looked into the issue further. But I've been using this since then and I haven't appreciated any side effects.

Reply:
@v-lopez 
First, thanks for the PR, we appreciate it.
However, after reading the origin thread I am a little confused, as it seems to mix ROS (initial_reset) and plain libRS where the fix was made.
Can you provide a way for this deadlock to be reproduced with librealsense, either in C++ or Python? Once I understand how to reproduce I can add it to our unit-tests and see if I can solve it. Otherwise we'll likely direct it to our ROS guy to try and reproduce and see if the problem is there.